### PR TITLE
staging-next 2023-07-28

### PR DESCRIPTION
**Pull request by vcunat**
##### Helpful links

- workflow docs: https://nixos.org/manual/nixpkgs/unstable/#submitting-changes-commit-policy
- important jobs:
  https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
  https://hydra.nixos.org/job/nixos/staging-next-small/tested
- jobsets:
  https://hydra.nixos.org/jobset/nixpkgs/staging-next
  https://hydra.nixos.org/jobset/nixos/staging-next-small
- nix-review reports: https://malob.github.io/nix-review-tools-reports/
- previous staging-next: https://github.com/NixOS/nixpkgs/pull/244111
- - - 
##### Mass breakages
(will be edited based on progress)

### CODE REVIEW

**Comment by stigtsp** [comment]
Looking at misc perlPackages failures

**Comment by doronbehar** [comment]
In a few minutes I'll push a small cleanup to pythran from https://github.com/NixOS/nixpkgs/pull/212913 . I will almost finish test that scipy builds with that change, along with merge conflicts fixed.

**Comment by doronbehar** [comment]
There is a problem with the hash of `confy` - that's the only merge conflict that should be resolved in `master`, see https://github.com/NixOS/nixpkgs/pull/246048 . It could be that upstream's archives creator is not deterministic.

**Comment by doronbehar** [comment]
I hope there won't be merge conflicts issues once https://github.com/NixOS/nixpkgs/pull/246584 is merged to master.

**Comment by stigtsp** [comment]
- https://github.com/NixOS/nixpkgs/pull/246585

File: pkgs/applications/misc/confy/default.nix [hunk 1]
```diff
@@ -34,7 +34,6 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
-    gobject-introspection
```
Comment (mweinelt) [inline pkgs/applications/misc/confy/default.nix:36]: Not removed from function arguments.

Comment (vcunat) [reply pkgs/applications/misc/confy/default.nix:36]: Because kept in `nativeBuildInputs`.

Comment (vcunat) [reply pkgs/applications/misc/confy/default.nix:36]: Because kept in `nativeBuildInputs`.

**Comment by vcunat** [comment]
- [ ] `python3Packages.numba` fails a version constraint after numpy update
  https://hydra.nixos.org/build/229681855/nixlog/2/tail
  Maintainer: @FRidh

**Comment by mweinelt** [comment]
Numba is  not adopting new numpy vereions fast enough. Not really something I would consider fixable right now.

**Comment by doronbehar** [comment]
Numba developers merged this 20 Hours ago:

https://github.com/numba/numba/pull/9105

Should we wait a day or two for a release? Or perhaps package an `unstable-` version?

**Comment by doronbehar** [comment]
Oh now I see that according to [their Github milestone](https://github.com/numba/numba/milestone/69), a release might be too far in the future.

**Comment by vcunat** [comment]
Uh, so
- ~(a)~ two numpy versions: probably not good; I suspect we'd end up with some python packages transitively depending on both versions  (I see combining scipy+numba at a quick glance)
- (b) downgrade numpy: I don't expect that to be good either
- (c) unstable numba version?  We're calling it "nixos-unstable" after all.  It should be better than a broken package, whether marked so or not.

**Comment by doronbehar** [comment]
> * unstable numba version?  We're calling it "nixos-unstable" after all.  It should be better than a broken package, whether marked so or not.

That's the PR I'm preparing right now.

**Comment by doronbehar** [comment]
https://github.com/NixOS/nixpkgs/pull/246727

**Comment by tjni** [comment]
@FRidh Just wanted to give you a heads up that python-acoustic's test fail after the Pandas 2 update.

It looks related to https://github.com/pandas-dev/pandas/issues/50051, and pandas has made some changes to how they do date parsing. I can't make heads or tails of it, and I noticed that the `cirrus` data files can contain a mixture of date formats (e.g. some dates are like m/dd/yyyy with the month first and not zero-padded, while others could be like dd/mm/yyyy with the month zero padded), so I'm not sure how to fix it.

**Comment by vcunat** [comment]
The numbers on Hydra still look pretty bad, at least 1.5k builds worse than nixpkgs master:
https://hydra.nixos.org/eval/1798274?compare=1798270#tabs-now-fail

I'd say that most of the regressions come from relatively lots of individual failures after updates of python packages.  A few hundred failures come from `python311Packages.rednose`, but that part seems the same as on master so _not_ a priority for this PR.

I'd really hope that we can merge in a better shape within a few days.  Thanks for your help with that.

**Comment by doronbehar** [comment]
- https://github.com/NixOS/nixpkgs/pull/247463 
- https://github.com/NixOS/nixpkgs/pull/247470
- https://github.com/NixOS/nixpkgs/pull/247471

**Comment by figsoda** [comment]
#247408

**Comment by doronbehar** [comment]
https://github.com/NixOS/nixpkgs/pull/247547

**Comment by vcunat** [comment]
Oh, I forgot to comment.  Thank you all for the numerous fixes.  Hundreds of builds remain regressed for sure, but I think we can take it to master now and (possibly) continue fixing in there.

**Pull request merged**